pkg/cmd/pipelinerun/delete_test.go

```diff
@@ -871,6 +887,52 @@ func TestPipelineRunDelete_v1beta1(t *testing.T) {
 			wantError:   true,
 			want:        "pipelineruns.tekton.dev \"nonexistent\" not found",
 		},
+		// test with pipelinerun have status as nil
+		{
+			name:        "Delete all the pipelineruns with keep-since except one without status conditions",
+			command:     []string{"delete", "--keep-since", "1", "-n", "ns", "-f"},
+			dynamic:     seeds[8].dynamicClient,
+			input:       seeds[8].pipelineClient,
+			inputStream: nil,
+			wantError:   false,
+			want:        "3 expired PipelineRuns(Completed) has been deleted in namespace \"ns\", kept 0\n",
+		},
+		{
+			name:        "Delete all the pipelineruns with keep-since except one without status conditions",
+			command:     []string{"delete", "--keep", "1", "-n", "ns", "-f", "--ignore-running=false"},
+			dynamic:     seeds[8].dynamicClient,
+			input:       seeds[8].pipelineClient,
+			inputStream: nil,
+			wantError:   false,
+			want:        "All but 1 PipelineRuns deleted in namespace \"ns\"\n",
+		},
+		{
+			name:        "Delete all the pipelineruns with keep-since ignore-running false except one without status conditions",
+			command:     []string{"delete", "--keep-since", "1", "-n", "ns", "-f", "--ignore-running=false"},
+			dynamic:     seeds[9].dynamicClient,
+			input:       seeds[9].pipelineClient,
+			inputStream: nil,
+			wantError:   false,
+			want:        "3 expired PipelineRuns has been deleted in namespace \"ns\", kept 0\n",
+		},
+		{
+			name:        "Delete all the pipelineruns including one without status",
+			command:     []string{"delete", "--all", "-n", "ns", "-f", "--ignore-running=false"},
+			dynamic:     seeds[9].dynamicClient,
+			input:       seeds[9].pipelineClient,
+			inputStream: nil,
+			wantError:   false,
+			want:        "All PipelineRuns deleted in namespace \"ns\"\n",
+		},
+		{
+			name:        "No pipelineruns after delete",
```

Comment:
maybe we can improve here and be more specific for the case we are checking for, maybe, "No pipelineruns after deleting running ones`, wdyt?